src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -0,0 +1,276 @@
+  { val := c • μ.val,
+    property := @measure_theory.finite_measure_smul_nnreal α _ μ.val μ.prop c, }), }
+
+-- TODO: A coercion `finite_measures α → measure α` is crucial. Now it is the subtype coercion.
+-- Another alternative is to use a bijection with `vector_measure α ℝ≥0` as an intermediate step.
+-- Potential advantages of the vector measure alternative:
+--  * Structures of additive commutative monoid and ℝ≥0 module on `finite_measure α` could be
+--    "inherited".
+-- Potential drawbacks of the vector measure alternative:
+--  * No integration theory directly.
+--  * Monotonicity lost as non-measurable sets defined to have measure 0.
+
+/-- A finite measure can be interpreted as a measure. -/
+instance : has_coe (finite_measures α) (measure_theory.measure α) := coe_subtype
+
+instance to_measure.finite_measure (μ : finite_measures α) :
+  finite_measure (μ : measure α) := μ.prop
+
+instance (α : Type*) [measurable_space α] :
+  has_coe_to_fun (finite_measures α) := ⟨(λ _, set α → ℝ≥0),
+    (λ μ, (λ s, (μ.val.measure_of s).to_nnreal))⟩
+
+lemma to_fun_eq_to_measure_to_nnreal (ν : finite_measures α) :
+  (ν : set α → ℝ≥0) = λ s, ((ν : measure α) s).to_nnreal := rfl
+
+lemma to_measure_eq_val (ν : finite_measures α) : (ν : measure α) = ν.val := rfl
+
+lemma coe_injective :
+  function.injective (coe : finite_measures α → measure α) :=
+subtype.coe_injective
+
+@[simp]
+lemma coe_zero : (coe : finite_measures α → measure α) 0 = 0 := rfl
+
+@[simp]
+lemma coe_add (μ ν : finite_measures α) :
+  (coe : finite_measures α → measure α) (μ + ν) = (μ : measure α) + (ν : measure α) := rfl
+
+@[simp]
+lemma coe_smul (c : ℝ≥0) (μ : finite_measures α) :
+  (coe : finite_measures α → measure α) (c • μ)  = c • (μ : measure α) := rfl
+
+/-- The (total) mass of a finite measure `μ` is `μ univ`, i.e., the cast to `nnreal` of
+`(μ : measure α) univ`. -/
+def mass {α : Type*} [measurable_space α] (μ : finite_measures α) : ℝ≥0 := μ univ
+
+lemma mass_def {α : Type*} [measurable_space α] {μ : finite_measures α} :
+  μ.mass = μ univ := rfl
+
+@[simp] lemma mass_ennreal {α : Type*} [measurable_space α]
+  {μ : finite_measures α} : (μ.mass : ℝ≥0∞) = (μ : measure α) univ :=
+begin
+  apply ennreal.coe_to_nnreal,
+  exact ennreal.lt_top_iff_ne_top.mp (μ.prop).measure_univ_lt_top,
+end
+
+-- TODO: Another alternative would use a bijection with `vector_measure α ℝ≥0`.
+instance {α : Type*} [measurable_space α] :
+  add_comm_monoid (finite_measures α) :=
+(finite_measures.coe_injective).add_comm_monoid
+  (coe : finite_measures α → measure α) finite_measures.coe_zero finite_measures.coe_add
+
+-- TODO: Another alternative would use a bijection with `vector_measure α ℝ≥0`.
+instance {α : Type*} [measurable_space α] :
+  module ℝ≥0 (finite_measures α) :=
+{ smul := (@finite_measures.has_scalar α _).smul,
+  one_smul := begin
+    intros μ,
+    apply finite_measures.coe_injective,
+    simp only [coe_smul, one_smul],
+  end,
```

Comment:
Same proof, more condensed (appropriate for a long series of completely unsurprising proofs, like you have here):
```suggestion
  one_smul := λ μ, finite_measures.coe_injective $ by simp only [coe_smul, one_smul],
```
Can you please do the same for the others?

Reply:
I would guess you can golf much further with `function.injective.module`

Reply:
@eric-wieser The problem is that in this situation we're making a `nnreal`-module as the pullback by an injective function of an `ennreal`-module -- the rings are different.

Reply:
I don't agree that there's a problem, I tried out `injective.module` and it worked just fine.

Reply:
Oh! I *thought* there would be a problem exactly as described by Heather --- I did notice `measure` had `has_scalar nnreal` in addition to `has_scalar ennreal`, but thought there is no way the `is_finite_measure` instance would be directly obtained.

This is a significant streamlining, thank you very much Eric!